code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -61,10 +63,13 @@ var/global/list/parasites = list() //all currently existing/living guardians
 		if("tech")
 			for(var/type in (subtypesof(/datum/guardianname/tech) - namedatum.type))
 				possible_names += new type
+		if("carp")
+			for(var/type in (subtypesof(/datum/guardianname/carp) - namedatum.type))
+				possible_names += new type
 	namedatum = pick(possible_names)
-	updatetheme()
+	updatetheme(pickedtheme)
 
-/mob/living/simple_animal/hostile/guardian/proc/updatetheme() //update the guardian's theme to whatever its datum is; proc for adminfuckery
+/mob/living/simple_animal/hostile/guardian/proc/updatetheme(theme) //update the guardian's theme to whatever its datum is; proc for adminfuckery
```

Comment:
You don't actually need to pass the theme for this; you can do `if(namedatum.theme == "carp")` instead